hydra-node/src/Hydra/API/HTTPServer.hs

```diff
@@ -72,7 +72,7 @@ instance Arbitrary DraftCommitTxResponse where
 -- witnessing via reference inputs
 data ScriptInfo = ScriptInfo
   { redeemer :: HashableScriptData
-  , datum :: HashableScriptData
+  , datumHash :: Maybe HashableScriptData
```

Comment:
I like that we do not need to change much to make this work!

The actual change of the `/commit` request body is a bit surprising though:
* This is a breaking change and I'm surprised the `api.yaml` did not need to be changed; I would have hoped the CI would pick this up
* Naming this `datumHash` indicates that this is the hash of a datum; But in fact, it is the pre-image (the datum itself) which needs to be given here. Keeping the previous name and making it `Maybe` would have been just fine.

I think examples would help to build the intuition here. We have examples in the api.yaml and I expect that we update them to have both scenarios covered.

Also, we wanted to fix the api reference description of the `UTxO` and `UTxOWithWitness` according to the issue description.

Reply:
You can't have examples that are not a valid json. I can add object keys, something like:

```
{ "example of request using inline datums" : {...},
  "example of request with datum in a transaction" : {...}
}
```
but it feels awkward. Maybe we should finally switch to open api if that provides better options for us?